assets/js/googlesitekit/notifications/datastore/notifications.js

```diff
@@ -181,7 +210,59 @@ export const actions = {
 	),
 };
 
-export const controls = {};
+export const controls = {
+	[ POPULATE_QUEUE ]: createRegistryControl(
+		( registry ) =>
+			async ( { payload } ) => {
+				const { viewContext, groupID } = payload;
+				const { isNotificationDismissed } =
+					registry.select( CORE_NOTIFICATIONS );
+				const notifications = registry
+					.select( CORE_NOTIFICATIONS )
+					.getNotifications();
+
+				// Wait for all dismissed items to be available before filtering.
+				await registry.resolveSelect( CORE_USER ).getDismissedItems();
+
+				let potentialNotifications = Object.values( notifications )
+					.filter(
+						( notification ) => notification.groupID === groupID
+					)
+					.filter( ( notification ) =>
+						notification.viewContexts.includes( viewContext )
+					)
+					.filter( ( { isDismissible, id } ) =>
+						isDismissible ? ! isNotificationDismissed( id ) : true
+					)
+					.map( ( { checkRequirements, ...notification } ) => ( {
+						...notification,
+						checkRequirements,
+						async check() {
+							if ( checkRequirements ) {
+								return await checkRequirements( registry );
+							}
+							return true;
+						},
```

Comment:
Async/await is redundant here. We can simply return the result of the `checkRequirements` function call (which is a promise) and it will have the same effect.
```suggestion
						check() {
							return checkRequirements
								? checkRequirements( registry )
								: Promise.resolve( true );
						},
```

Reply:
That's right, some of the `checkRequirements` callbacks are not async, e.g.:

https://github.com/google/site-kit-wp/blob/b5816b5701f02da1b6ab099e83467719e7b7e63f/assets/js/modules/ads/index.js#L108-L144

Reply:
Oh, good call. Thanks.